logger-kafka-consumer/src/main/java/com/trihydro/loggerkafkaconsumer/app/services/TimRsuService.java

```diff
@@ -1,69 +1,75 @@
+     */
     public Long AddTimRsu(Long timId, Integer rsuId, Integer rsuIndex) {
+        String insertQueryStatement = timDbTables.buildInsertQueryStatement("tim_rsu", timDbTables.getTimRsuTable());
 
-        PreparedStatement preparedStatement = null;
-        Connection connection = null;
-
-        try {
-            connection = dbInteractions.getConnectionPool();
-            String insertQueryStatement = timDbTables.buildInsertQueryStatement("tim_rsu",
-                    timDbTables.getTimRsuTable());
-            preparedStatement = connection.prepareStatement(insertQueryStatement, new String[] { "tim_rsu_id" });
+        try (Connection connection = dbInteractions.getConnectionPool(); PreparedStatement preparedStatement = connection.prepareStatement(insertQueryStatement, new String[] {"tim_rsu_id"})) {
             int fieldNum = 1;
 
             for (String col : timDbTables.getTimRsuTable()) {
-                if (col.equals("TIM_ID"))
+                if (col.equals("TIM_ID")) {
                     sqlNullHandler.setLongOrNull(preparedStatement, fieldNum, timId);
-                else if (col.equals("RSU_ID"))
+                } else if (col.equals("RSU_ID")) {
                     sqlNullHandler.setIntegerOrNull(preparedStatement, fieldNum, rsuId);
-                else if (col.equals("RSU_INDEX"))
+                } else if (col.equals("RSU_INDEX")) {
                     sqlNullHandler.setIntegerOrNull(preparedStatement, fieldNum, rsuIndex);
+                }
                 fieldNum++;
             }
-            Long timRsuId = dbInteractions.executeAndLog(preparedStatement, "tim rsu");
-            return timRsuId;
+            return dbInteractions.executeAndLog(preparedStatement, "tim rsu");
         } catch (SQLException e) {
-            // java.sql.SQLIntegrityConstraintViolationException: ORA-00001: unique constraint (CVCOMMS.TIM_RSU_U) violated 
-            if(!(e instanceof SQLIntegrityConstraintViolationException)) {
-                e.printStackTrace();
+            if (e.getMessage() != null && e.getMessage().contains("unique constraint")) {
+                // Avoid logging the common error when trying to insert a duplicate record
+                return 0L;
             }
 
-            return Long.valueOf(0);
-        } finally {
-            try {
-                // close prepared statement
-                if (preparedStatement != null)
-                    preparedStatement.close();
-                // return connection back to pool
-                if (connection != null)
-                    connection.close();
-            } catch (SQLException e) {
-                e.printStackTrace();
-            }
+            // Log the exception if it's not a unique constraint violation
+            log.error("SQL Exception while adding TIM RSU: {}", e.getMessage(), e);
+            return -1L;
         }
```

Comment:
Note: Returning different long values here makes it easier to verify various code paths through unit tests.